pkg/test/path.go

```diff
@@ -0,0 +1,45 @@
+package test
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+)
+
+// Return the absolute path of the repository.
+func RepoAbsPath() (string, error) {
+	_, file, _, ok := runtime.Caller(0)
```

Comment:
If we agree that patterns like:
```go
myfile := "../../foo/bar"
```
are bad because they are subject to the volatile value of the running processes's cwd, then it follows we need another way.

In researching on the Internet, I found that Golang doesn't have an analog for the C `__FILE__` macro, from which several other languages derive the same convention (like Python's `__file__`).

In my findings (limited, not extensive), the answer came back to using this. I will also concede that I didn't do any research to understand when `ok` might return false.

So to me the options were only:

1. Continue with the bad pattern as written above
1. Use this

I am happy to hear if there are other options my research missed or if this can be improved.

Reply:
"go test" runs a package's tests from within the package directory. (The documentation for this hides near the bottom of the output from "go help testflag".)

The convention in tests is to rely on this by using paths that are relative to the package directory.

Reply:
Done at 1d8941b .